mne/io/edf/edf.py

```diff
@@ -511,16 +481,529 @@ def _get_edf_info(fname, stim_channel, annot, annotmap, eog, misc, exclude,
+            number, as a combination of 'unit code' + order of magnitude.
+            The list of possible codes can be found in the following
+            documents:
+            - Units codes :
+            https://sourceforge.net/p/biosig/svn/HEAD/tree/trunk/biosig/doc/units.csv
+            - Decimal factors codes:
+            https://sourceforge.net/p/biosig/svn/HEAD/tree/trunk/biosig/doc/DecimalFactors.txt
+
+            For example, the physical dimension 'uV' (microvolt) is
+            encoded as 4256 (Volt) + 19 (micro) = 4275. The code 512
+            stands for dimensionless.
+            """  # noqa
+            fid.seek(6 * len(channels), 1)  # phys_dim, obsolete
+            units = np.fromfile(fid, np.uint16, len(channels)).tolist()
+            unitcodes = np.array(units[:])
+            include = list()
+            for i, unit in enumerate(units):
+                if unit == 4275:  # microvolts
+                    units[i] = 1e-6
+                elif unit == 512:
+                    units[i] = 1
+                else:
+                    warn('Unknown physical dimension for channel %d. '
+                         'Assuming dimensionless.' % i)
+                    units[i] = 1
+                include.append(i)
+
+            ch_names = [ch_names[idx] for idx in include]
+            physical_min = np.fromfile(fid, np.float64, len(channels))
+            physical_max = np.fromfile(fid, np.float64, len(channels))
+            digital_min = np.fromfile(fid, np.float64, len(channels))
+            digital_max = np.fromfile(fid, np.float64, len(channels))
+
+            fid.seek(68 * len(channels), 1)  # obsolete
+
+            lowpass = np.fromfile(fid, np.float32, len(channels))
+            highpass = np.fromfile(fid, np.float32,  len(channels))
+            notch = np.fromfile(fid, np.float32, len(channels))
+
+            # number of samples per record
+            n_samps = np.fromfile(fid, np.int32, len(channels))
+
+            # data type
+            gdftype = np.fromfile(fid, np.int32, len(channels))
+
+            channel = {}
+            channel['xyz'] = [np.fromfile(fid, np.float32, 3)[0]
+                              for ch in channels]
+
+            if edf_info['number'] < 2.19:
+                impedance = np.fromfile(fid, np.uint8, len(channels))
+                channel['impedance'] = pow(2, impedance.astype(float) / 8)
+                fid.seek(19 * len(channels), 1)  # reserved
+            else:
+                tmp = np.fromfile(fid, np.float32, 5 * len(channels))
+                tmp = tmp[::5]
+                fZ = tmp[:]
+                impedance = tmp[:]
+                # channels with no voltage (code 4256) data
+                ch = [unitcodes & 65504 != 4256][0]
+                impedance[np.where(ch)] = None
+
+                # channel with no impedance (code 4288) data
+                ch = [unitcodes & 65504 != 4288][0]
+                fZ[np.where(ch)[0]] = None
```

Comment:
> Can you compare to the values you get in some other library like EEGLAB or FieldTrip at least? We should establish some confidence that we're getting correct values before merging this.

I have tested this using fieldtrip's biosig actually, and the values are the same. 

But as I said my files I don't actually use this in my analysis so I don't know if the values given by fieldtrip make any sense either.